controllers/integration/integration_adapter.go

```diff
@@ -90,10 +119,43 @@ func (a *Adapter) EnsureApplicationSnapshotExists() (results.OperationResult, er
 		return results.RequeueOnErrorOrStop(a.updateStatus())
 	}
 
+	newIntegrationTestScenarios, err := a.getAllIntegrationTestScenariosForApplication(a.application)
```

Comment:
nit: I think we should rename `newIntegrationTestScenarios` to `allIntegrationTestScenarios` or similar as they are not new scenarios, but existing ones